unitary/examples/quantum_rpg/bb84.py

```diff
@@ -0,0 +1,211 @@
+# Copyright 2023 The Unitary Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Tuple
+import random
+
+
+from unitary.examples.quantum_rpg.game_state import GameState
+from unitary.examples.quantum_rpg.item import EXAMINE, TALK, Item
+from unitary.examples.quantum_rpg.world import Direction, Location
+
+
+_WORDS = [
+    "QEC",
+    "ket",
+    "Arc",
+    "sum",
+    "sin",
+    "cos",
+    "QED",
+    "log",
+    "add",
+    "BQP",
+    "HOG",
+    "QFT",
+    "QKD",
+    "VQE",
+    "RSA",
+]
+_BASIS = ["I", "H"]
+_VALUES = ["0", "1"]
+
+
+def to_bin(c: str) -> str:
+    """Converts a single character to a binary ASCII string.
+
+    This returns a str of length 8 that corresponds to the
+    ASCII value of the character.
+    """
+    return ("00000000" + bin(ord(c)).replace("0b", ""))[-8:]
+
+
+def gen_bb84(word: str) -> Tuple[str, str]:
+    alice = ""
+    bob = ""
+    key = "".join(to_bin(c) for c in word)
+    c = 0
+    while c < len(key):
+        alice_base = random.choice(_BASIS)
+        bob_base = random.choice(_BASIS)
+        if alice_base == bob_base:
+            alice_sent = key[c]
+            bob_recv = key[c]
+            c += 1
+        else:
+            alice_sent = random.choice(_VALUES)
+            bob_recv = random.choice(_VALUES)
+        alice += alice_base + alice_sent
+        bob += bob_base + bob_recv
+    return (alice, bob)
+
+
+def solve_bb84(alice: str, bob: str):
+    """Hey, no cheating!
+
+    This solves a alice/bob BB84 riddle.  Used for testing
+    or for if you get stuck.
+    """
+
+    assert len(alice) == len(bob), "Alice or Bob are missing bits!"
+    key = ""
+    for idx in range(0, len(alice), 2):
+        if alice[idx] == bob[idx]:
+            assert alice[idx + 1] == bob[idx + 1], "Eve has measured a qubit!"
+            key += alice[idx + 1]
+    word = ""
+    for idx in range(0, len(key), 8):
+        word += chr(int(key[idx : idx + 8], 2))
+    return word
+
+
+def _gen_keys(state: GameState):
+    if "akey" not in state.state_dict:
```

Comment:
`akey` is never set anywhere. Should this be `alice`/`bob`?

Reply:
Fixed, but in #118 
This PR got screwed up somehow.